src/tool/EventTool.cpp

```diff
@@ -187,26 +196,76 @@ void EventTool::copyAction()
                 OffEvent* offEv = dynamic_cast<OffEvent*>(onEv->offEvent()->copy());
                 if (offEv) {
                     offEv->setOnEvent(onEv);
-                    copiedEvents->append(offEv);
+                    //copiedEvents->append(offEv);
+                    offEv->setTrack(copyTrack, false);
+                    offEv->setChannel(0, false);
+                    offEv->setFile(&copyFile);
+                    copyFile.channelEvents(0)->insert(offEv->midiTime(), offEv);
                 }
             }
         }
-        _mainWindow->copiedEventsChanged();
+        
+        QByteArray copyFileBytes = copyFile.toByteArray();
+        int MAX_COPY_SIZE = 1073741824; //1GB
+        if (copyFileBytes.size() > MAX_COPY_SIZE){
```

Comment:
Why do we have any MAX_COPY_SIZE? Is it a restriction of QSharedMemory or something?

Reply:
We should create buffer of some size, so we should set some size, MAX_COPY_SIZE in our case.

Reply:
The size of QSharedMemory needs to be specified before allocation, as far as i know. One can of course argue about what this size should be...